File: python/백준/자료구조/스택, 집합/17608 막대기.py
import sys

input = sys.stdin.readline

N = int(input())

stack = []
stack.append(int(input()))

for _ in range(N-1):
    num = int(input())
    while stack and stack[-1] <= num:
        stack.pop()
    
    stack.append(num)

print(len(stack))

File: cpp/백준/자료구조/스택, 집합/17608 막대기.cpp
#include <iostream>
#include <stack>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N;
    cin >> N;

    stack<int> st;
    int first;
    cin >> first;
    st.push(first);

    for (int i = 0; i < N - 1; ++i) {
        int num;
        cin >> num;
        while (!st.empty() && st.top() <= num) st.pop();
        st.push(num);
    }

    cout << st.size() << "\n";
    return 0;
}
